File: Data Structures & Algorithms/partition-labels/submission-0.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        lastIdx = {}

        for i in range(len(s)):
            lastIdx[s[i]] = i

        res = []
        size = 0
        end = 0

        for i in range(len(s)):
            size += 1
            end = max(end, lastIdx[s[i]])

            if i == end:
                res.append(size)
                size = 0
        return res

File: Data Structures & Algorithms/merge-intervals/submission-0.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        List<int[]> res = new ArrayList<>();
        res.add(intervals[0]);

        for(int[] interval : intervals){
            int start = interval[0];
            int end = interval[1];
            int lastEnd = res.get(res.size() - 1)[1];

            if (interval[0] <= res.get(res.size() - 1)[1]){
                res.get(res.size() - 1)[1] = Math.max(end, lastEnd);
            } else {
                res.add(interval);

            }
        }
        return res.toArray(new int[res.size()][]);
    }
}
